lib/icinga/pluginutility.cpp

```diff
@@ -38,11 +39,29 @@ void PluginUtility::ExecuteCommand(const Command::Ptr& commandObj, const Checkab
 	const std::function<void(const Value& commandLine, const ProcessResult&)>& callback)
 {
 	Value raw_command = commandObj->GetCommandLine();
-	Dictionary::Ptr raw_arguments = commandObj->GetArguments();
+	Value raw_plugin = commandObj->GetPlugin();
 
 	Value command;
+	Dictionary::Ptr raw_arguments = commandObj->GetArguments();
 
 	try {
+		if (!raw_plugin.IsEmpty()) {
+			Value plugin = LookupPlugin(raw_plugin);
+
+			if (!plugin.IsEmpty()) {
+				if (raw_command.IsEmpty())
+					raw_command = plugin;
+				else if (raw_command.IsObjectType<Array>())
+					raw_command << plugin;
```

Comment:
The `<<` (bitwise left-shift) operator does not do what you probably want it to do here.